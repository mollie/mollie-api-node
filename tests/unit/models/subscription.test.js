import Subscription from 'models/subscription';

describe('subscription model', () => {
  it('should instantiate with default values', () => {
    const subscription = new Subscription();

    expect(subscription.isActive()).toBe(false);
    expect(subscription.isPending()).toBe(false);
    expect(subscription.isCanceled()).toBe(false);
    expect(subscription.isSuspended()).toBe(false);
    expect(subscription.isCompleted()).toBe(false);
    expect(subscription.getWebhookUrl()).toBeNull();
    expect(subscription.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const subscriptionProps = JSON.parse(`{
        "resource": "subscription",
        "id": "sub_rVKGtNd6s3",
        "mode": "live",
        "createdAt": "2016-06-01T12:23:34+00:00",
        "status": "active",
        "amount": {
            "value": "25.00",
            "currency": "EUR"
        },
        "times": 4,
        "interval": "3 months",
        "description": "Quarterly payment",
        "method": null,
        "webhookUrl": "https://webshop.example.org/payments/webhook",
        "_links": {
            "self": {
                "href": "https://api.mollie.com/v2/customers/cst_stTC2WHAuS/subscriptions/sub_rVKGtNd6s3",
                "type": "application/hal+json"
            },
            "customer": {
                "href": "https://api.mollie.com/v2/customers/cst_stTC2WHAuS",
                "type": "application/hal+json"
            },
            "documentation": {
                "href": "https://docs.mollie.com/reference/v2/subscriptions-api/get-subscription",
                "type": "text/html"
            }
        }
    }`);
    const subscription = new Subscription(subscriptionProps);

    expect(subscription.isActive()).toBe(true);
    expect(subscription.isPending()).toBe(false);
    expect(subscription.isCanceled()).toBe(false);
    expect(subscription.isSuspended()).toBe(false);
    expect(subscription.isCompleted()).toBe(false);
    expect(subscription.getWebhookUrl()).toBe(subscriptionProps.webhookUrl);
    expect(subscription.toPlainObject()).toMatchSnapshot();
  });
});
