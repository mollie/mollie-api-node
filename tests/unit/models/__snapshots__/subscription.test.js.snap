// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscription model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "customer": null,
  },
  "amount": Object {
    "currency": null,
    "value": null,
  },
  "canceledAt": null,
  "createdAt": null,
  "description": null,
  "id": null,
  "interval": null,
  "method": null,
  "mode": null,
  "resource": "subscription",
  "startDate": null,
  "status": null,
  "times": null,
  "webhookUrl": null,
}
`;

exports[`subscription model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "customer": Object {
      "href": "https://api.mollie.com/v2/customers/cst_stTC2WHAuS",
      "type": "application/hal+json",
    },
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/subscriptions-api/get-subscription",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/customers/cst_stTC2WHAuS/subscriptions/sub_rVKGtNd6s3",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "EUR",
    "value": "25.00",
  },
  "canceledAt": null,
  "createdAt": "2016-06-01T12:23:34+00:00",
  "description": "Quarterly payment",
  "id": "sub_rVKGtNd6s3",
  "interval": "3 months",
  "method": null,
  "mode": "live",
  "resource": "subscription",
  "startDate": null,
  "status": "active",
  "times": 4,
  "webhookUrl": "https://webshop.example.org/payments/webhook",
}
`;
